packages/xod-arduino/src/transpiler.js

```diff
@@ -121,12 +121,17 @@ const formatValueLiteral = def(
 
 // Creates a TConfig object from entry-point path and project
 const createTConfig = def(
-  'createTConfig :: PatchPath -> Project -> TConfig',
-  (path, project) => R.applySpec({
-    NODE_COUNT: R.compose(getNodeCount, Project.getPatchByPathUnsafe(path)),
-    MAX_OUTPUT_COUNT: getOutputCount,
-    XOD_DEBUG: R.F,
-  })(project)
+  'createTConfig :: PatchPath -> [TNodeId] -> Project -> TConfig',
+  (path, deferredNodeIds, project) => R.compose(
```

Comment:
You haven’t to pass the `deferredNodeIds` if toposort them to the tail. Counting could be done right inside the function implementation.